packages/library/src/brokers/rabbit/main.ts

```diff
@@ -70,40 +95,96 @@ export default class Rabbit extends MessageBroker {
     this.connection = null;
     this.channel = null;
     const that = this;
-    (async () => {
-      that.connection = await amqp.connect(defaultConfig);
-      if (that.connection === null)
-        throw new Error('No connection for Rabbit.');
-
-      that.channel = await that.connection!.createChannel();
-      if (that.channel === null) throw new Error('No channel for Rabbit.');
-
-      // const that = that;
-      Object.keys(topics).forEach(async (topic) => {
-        // POSSIBLE REFACTOR: Don't know if we need await
-        await that.channel?.assertExchange(
-          that.topics[topic].exchange.name,
-          that.topics[topic].exchange.type ?? 'topic',
-          that.topics[topic].exchange ?? {}
-        );
-
-        await that.channel?.assertQueue(topic, that.topics[topic] ?? {});
-
-        // TODO: research and implement "args"
-        await that.channel?.bindQueue(
-          topic,
-          that.topics[topic].exchange.name,
-          that.topics[topic].key ?? topic
-        );
-      });
-    })();
+
+    // (async () => {
+    //   that.connection = await amqp.connect(defaultConfig);
+    //   if (that.connection === null)
+    //     throw new Error('No connection for Rabbit.');
+
+    //   that.channel = await that.connection!.createChannel();
+    //   if (that.channel === null) throw new Error('No channel for Rabbit.');
+
+    //   // console.log('this is the channel', that.channel);
+    //   // const that = that;
+
+    //   console.log('lets look at the new topics', that.topics);
+    //   Object.keys(topics).forEach(async (topic) => {
+    //     // POSSIBLE REFACTOR: Don't know if we need await
+    //     await that.channel?.assertExchange(
+    //       that.topics[topic].exchange.name,
+    //       that.topics[topic].exchange.type ?? 'topic',
+    //       that.topics[topic].exchange ?? {}
+    //     );
+
+    //    const q = await that.channel?.assertQueue(topic, that.topics[topic] ?? {});
+
+    //    console.log('this is the created q', q);
+
+    //     // TODO: research and implement "args"
+    //     await that.channel?.bindQueue(
+    //       topic,
+    //       that.topics[topic].exchange.name,
+    //       that.topics[topic].key ?? topic
+    //     );
+    //   });
+
+    console.log('in constructor - channel', that.channel);
+
+    console.log('in constructor - connection', that.connection);
+
+    console.log('this is that in the constructor', that);
```

Comment:
If you don't mind removing zombie/debug code.

Reply:
Removed from the updated code.